### ISSUE TITLE
The test results were terrible.

### ISSUE THREAD

**Issue by yumengWang112** [fenglinglwb/MAT#26]
Thank you so much for a great job. I had a few problems when I was testing. I made a few minor changes to the code, which worked fine in training, but in testing, it printed out a lot of intermediate parameters, probably executed the code(fig3), and the test result were terrible. I want to know do you have encountered this problem and how to solve it?
![5770e7be022e402c7b3ace9ffec584c](https://user-images.githubusercontent.com/79405379/183244881-3c969ce6-7171-4e98-98e4-08a49ce416c6.jpg)
![图片](https://user-images.githubusercontent.com/79405379/183245090-19347b99-b436-4f66-81b1-77476c12d983.png)

![22402d3480e7a06f193c2ea680b9997](https://user-images.githubusercontent.com/79405379/183245097-2f3dbcd8-9000-45bd-a84e-00961d2cbb6d.jpg)
fig1is the test result   and the  fig2  is the snapshot of training.   Thanks a lot!

**Comment by yumengWang112** [comment]
I'm sorry that i just saw that you updated the generate_image.py file.
But the test results with this file were still terrible. 
I would like to ask what caused this bad result. Is it possible that the trained model or the test file needs to be modified?

**Comment by yumengWang112** [comment]
I'm sorry that i just saw that you updated the generate_image.py file.
But the test results with this file were still terrible.
I would like to ask what caused this bad result. Is it possible that the trained model or the test file needs to be modified?
I am very confused. because the snapshot of training perfform well.

**Comment by fenglinglwb** [comment]
The printed information indicates an unmatching of the built network in 'generate_image.py' and the loaded one. If you have modified the network, you should redefine the _Generator_ in https://github.com/fenglinglwb/MAT/blob/e5b9ca73d2bee044d8c546c7d99200deedc7fd17/generate_image.py#L27.

**Comment by yumengWang112** [comment]
> 

Ok,I will have a try,Thanks a lot!

**Comment by yumengWang112** [comment]
Same problem as last time. I just added a loss function to the original model and did not change the generated network, but the message printed mapping.fc2-fc7. Weight and bias means that the saved model does not match these parameters of the generator. I have been confused for a long time but I still can't solve this problem. I would like to consult you，Looking forward to your reply!



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "fenglinglwb/MAT"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年8月6日(星期六) 晚上11:22
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [fenglinglwb/MAT] The test results were terrible. (Issue #26)





 
The printed information indicates an unmatching of the built network in 'generate_image.py' and the loaded one. If you have modified the network, you should redefine the Generator in https://github.com/fenglinglwb/MAT/blob/e5b9ca73d2bee044d8c546c7d99200deedc7fd17/generate_image.py#L27.

**Comment by fenglinglwb** [comment]
I cannot tell what's wrong according to your provided information. You can send an email to wenboli@cse.cuhk.edu.hk and we can discuss the problem in detail.

**Comment by Fuze5k** [comment]
After a few hours of work I found the solution for this problem.
If the model was trained on png images you should also use png images in your training set. 
If the model was trained on jpg images, you should also use jpg images in your training set.
